db/blob/blob_source.cc

```diff
@@ -202,32 +202,30 @@ Status BlobSource::GetBlob(const ReadOptions& read_options,
   return s;
 }
 
-void BlobSource::MultiGetBlob(
-    const ReadOptions& read_options,
-    std::map<uint64_t, autovector<BlobReadRequest>>& blob_reqs,
-    uint64_t* bytes_read) {
+void BlobSource::MultiGetBlob(const ReadOptions& read_options,
+                              autovector<BlobFileReadRequests>& blob_reqs,
+                              uint64_t* bytes_read) {
   assert(blob_reqs.size() > 0);
 
   Status s;
 
-  for (auto& batch_rqs : blob_reqs) {
-    const uint64_t file_number = batch_rqs.first;
-    CacheHandleGuard<BlobFileReader> blob_file_reader;
-    s = GetBlobFileReader(file_number, &blob_file_reader);
-    assert(!s.ok() || blob_file_reader.GetValue());
+  for (auto& batch_req : blob_reqs) {
+    auto& [file_number, file_size, blob_reqs_in_file] = batch_req;
```

Comment:
We could move the structured binding into the `for`

Reply:
Done